src/mono/wasm/test-main.js

```diff
@@ -42,7 +43,12 @@ function proxyMethod(prefix, func, asJson) {
             isXUnitDoneCheck = true;
         }
 
-        if (asJson) {
+        if (payload.length > 500 && payload.indexOf("STARTRESULTXML") !== -1) {
```

Comment:
`payload` would start with the `STARTRESULTXML` string, correct? If so, then it would be better to use `startsWith` so we don't have to scan the entire payload. Also, then we can skip the length check.